caching-service/src/main/java/org/zowe/apiml/caching/service/vsam/RemoveOldestStrategy.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.caching.service.vsam;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.zowe.apiml.caching.service.EvictionStrategy;
+import org.zowe.apiml.caching.service.vsam.config.VsamConfig;
+import org.zowe.apiml.zfile.ZFileConstants;
+import org.zowe.apiml.zfile.ZFileException;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Optional;
+
+
+@RequiredArgsConstructor
+@Slf4j
+public class RemoveOldestStrategy implements EvictionStrategy {
+    private final VsamConfig vsamConfig;
+
+    private final VsamFile file;
+
+    @Override
+    public void evict(String key) {
+        removeOldestRecord();
+    }
+
+    public void removeOldestRecord() {
```

Comment:
Why is that public?

Reply:
You're right, it should be private, I will change it